src/Functions/Public/Connect-vRAServer.ps1

```diff
@@ -85,162 +99,170 @@
+                    $Params.Add("SkipCertificateCheck", $true)
 
-                $Params.Add("SslProtocol", $SslProtocol)
+                }
 
-            }
+                if (($SslProtocolResult -ne 'Default') -and ($IsCoreCLR)) {
 
-            $Response = Invoke-RestMethod @Params
+                    $Params.Add("SslProtocol", $SslProtocol)
 
-            if ('refresh_token' -in $Response.PSobject.Properties.Name) {
-                $Token = $Response.access_token
-                $RefreshToken = $Response.refresh_token
-            }
+                }
 
-            if ('token' -in $Response.PSobject.Properties.Name) {
-                $Token = $Response.token
-                $RefreshToken = $APIToken
-            }
+                $Response = Invoke-RestMethod @Params
 
-            # --- Create Output Object
-            $Script:vRAConnection = [PSCustomObject] @{
+                if ('refresh_token' -in $Response.PSobject.Properties.Name) {
+                    $Token = $Response.access_token
+                    $RefreshToken = $Response.refresh_token
+                }
 
-                Server = "https://$($Server)"
-                Token = $Token
-                RefreshToken = $RefreshToken
-                APIVersion = $Null
-                SignedCertificates = $SignedCertificates
-                SslProtocol = $SslProtocolResult
-            }
+                if ('token' -in $Response.PSobject.Properties.Name) {
+                    $Token = $Response.token
+                    $RefreshToken = $APIToken
+                }
+
+                # --- Create Output Object
+                $Script:vRAConnection = [PSCustomObject] @{
+
+                    Server = "https://$($Server)"
+                    Token = $Token
+                    RefreshToken = $RefreshToken
+                    APIVersion = $Null
+                    SignedCertificates = $SignedCertificates
+                    SslProtocol = $SslProtocolResult
+                }
 
-            # --- Update vRAConnection with API version
-            $Script:vRAConnection.APIVersion = (Get-vRAAPIVersion).APIVersion
+                # --- Update vRAConnection with API version
+                $Script:vRAConnection.APIVersion = (Get-vRAAPIVersion).APIVersion
 
-        }
-        catch [Exception]{
+            }
+            catch [Exception]{
+
+                if ($_.Exception.Response.StatusCode -eq "BadRequest") {
+                    # This could be due to an incorrectly set UserAttribute, so customizing the message.
+                    if ($Username -match "@") {
+                        Write-Host -ForegroundColor White -BackgroundColor DarkGray "Unfortunately, you have received a 400 BAD Request Error. Is it possible that your environment has userPrincipalName set as the Username attribute? If so, please make sure to include the parameter UserAttribute and set it to either UPN or userPrincipalName (e.g. -UserAttribute UPN)"
```

Comment:
This should all be part of the throw message, or (my preference) Write-Error with an -ErrorAction Stop.  By using Write-Host it's now impossible for an upstream script to capture the error.

Reply:
Good note, update incoming.

Reply:
updated here: [new commit](https://github.com/jakkulabs/PowervRA/pull/235/commits/c6c50a2d5cb03f45234dc7e67e4c3ea169acf543)